kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/util.kt

```diff
@@ -48,7 +49,8 @@ internal val ktSymbolOriginToOrigin = mapOf(
     KtSymbolOrigin.JAVA_SYNTHETIC_PROPERTY to Origin.SYNTHETIC,
     KtSymbolOrigin.INTERSECTION_OVERRIDE to Origin.KOTLIN,
     // TODO: distinguish between kotlin library and java library.
-    KtSymbolOrigin.LIBRARY to Origin.JAVA_LIB
+    KtSymbolOrigin.LIBRARY to Origin.KOTLIN_LIB,
+    KtSymbolOrigin.SUBSTITUTION_OVERRIDE to Origin.JAVA_LIB
```

Comment:
You can't tell the origin of (type) substituted override unless you check the original symbol.

See the example I added at https://github.com/JetBrains/intellij-community/commit/0c4ec1476ec22cab5ee8fbfb50fb535ddc3073af. The fake override added to `SubBox` where type parameter `t` is substituted with `String` will have `SUBSTITUTION_OVERRIDE`. And both of `SubBox` and `Box<T>` are in kotlin source in this case.